llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapInputFormat.java

```diff
@@ -137,6 +137,8 @@
       // This starts the reader in the background.
       rr.start();
       return result;
+    } catch (IOException ioe) {
```

Comment:
is this needed? Exception should catch everything right?

Reply:
Hey @pgaref,

Ya, this is required.  Based on the `InvalidInputException` (which is a subclass of `IOException`) changes in HDFS, this code is require to pass the `InvalidInputException` up to the caller directly, otherwise, in the `Exception` block, it gets wrapped in yet another `IOException` and that caller is no longer able to detect the `InvalidInputException`.

I hope that makes sense.